pkg/templates/templates.go

```diff
@@ -140,30 +143,28 @@ type EncryptionConfig struct {
 // TemplateResolver is the API for processing templates. It's better to use the NewResolver function
 // instead of instantiating this directly so that configuration defaults and validation are applied.
 type TemplateResolver struct {
-	kubeClient *kubernetes.Interface
-	kubeConfig *rest.Config
-	config     Config
-	// Denotes if the lookup template function encountered an API resource that is not installed on
-	// the Kubernetes API server.
-	missingAPIResource bool
-	referencedObjects  []client.ObjectIdentifier
+	config Config
+	// Used when caching is disabled.
+	dynamicClient *dynamic.DynamicClient
+	kubeConfig    *rest.Config
+	// Used when instantiated with NewResolverWithCaching. This will create watches and the cache will get
+	// automatically updated.
+	dynamicWatcher client.DynamicWatcher
+	// If caching is disabled, this will act as a temporary cache for objects during the execution of the
+	// ResolveTemplate call.
+	objectCache client.ObjectCache
```

Comment:
Is it correct that when "caching is disabled" the `objectCache` here will be used, but otherwise, this cache is not used, and the cache inside the `dynamicWatcher` is used?

Maybe that could be a little bit clearer in the doc strings here, or maybe in the name of the `objectCache` here - perhaps something like `tempCallCache`

Reply:
I realized these fields aren't exposed, so the docs aren't as necessary for users.

Reply:
I like the suggestion of renaming it. I'll fix that.